src/java/org/apache/cassandra/index/sai/disk/v1/V1SearchableIndex.java

```diff
@@ -169,6 +175,12 @@ public List<RangeIterator<Long>> search(Expression expression,
     {
         List<RangeIterator<Long>> iterators = new ArrayList<>();
 
+        if(keyRange instanceof Bounds && keyRange.left == keyRange.right && keyRange.left instanceof DecoratedKey)
```

Comment:
`keyRange.left == keyRange.right`

What is the reason for checking reference, and not object, equality here?